### PR TITLE
feat(theme): add theming solution

### PR DESCRIPTION
**Pull request by hpierre74**

### CODE REVIEW

File: src/Theme/ThemeContext.re [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  primary: string,
+  secondary: string,
+  accent: string,
+  textPrimary: string,
+  button: buttonTypes,
+};
+
+let theme = {
+  mode: "dark",
+  primary: "000088",
```
Comment (cchampou) [inline src/Theme/ThemeContext.re]: Go Variant :)

File: src/Button/ButtonStyles.ml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+open Emotion
+
+let default ~theme = [%css [
```
Comment (cchampou) [inline src/Button/ButtonStyles.ml]: Why not in Reason 😊

File: src/Theme/ThemeContext.re [hunk 3]
```diff
@@ -0,0 +1,28 @@
+type buttonTypes = {
+  disabled: string,
```
Comment (cchampou) [inline src/Theme/ThemeContext.re]: Boolean ?

File: src/Theme/ThemeContext.re (hunk 1 above)
Comment (hpierre74) [reply src/Theme/ThemeContext.re]: je test là c'est en wip^^

File: src/Theme/ThemeContext.re (hunk 3 above)
Comment (hpierre74) [reply src/Theme/ThemeContext.re]: it's the color

Comment (hpierre74) [reply src/Theme/ThemeContext.re]: maybe it should be more explicit, but since it was theming i didn't really thought of it

**Pull request merged**